Add Apache Kafka

vordimous opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

https://github.com/semantalytics/awesome-kafka

This list is about kafka which is an open-source distributed event streaming platform used by many teams to manage stateful events.

PRs Reviewed:
https://github.com/sindresorhus/awesome/pull/2622
https://github.com/sindresorhus/awesome/pull/2607

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- [x] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- [x] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [x] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions. Comments pointing out lint violation are allowed, but does **not** count as a review.
- [x] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- [x] This pull request has a title in the format `Add Name of List`. It should not contain the word `Awesome`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- [x] Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- [x] Your entry should be added at the bottom of the appropriate category.
- [x] The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- [x] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [x] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [x] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- [x] The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- [x] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- [x] It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- [x] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- [x] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- [x] Non-generated Markdown file in a GitHub repo.
- [x] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [x] Not a duplicate. Please search for existing submissions.
- [x] Only has awesome items. Awesome lists are curations of the best, not everything.
- [x] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- [x] Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- [x] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [x] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- [x] Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- [x] Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- [x] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- [x] All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- [x] Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [x] Does not use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- [x] Does not include a CI (e.g. GitHub Actions) badge.<br>You can still use a CI for linting, but the badge has no value in the readme.
- [x] Does not include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

asapelkin commented:
Hello, @vordimous! It looks like you need to fix some [awesome-lint](https://github.com/sindresorhus/awesome-lint ) errors first. 
-> % npx awesome-lint https://github.com/semantalytics/awesome-kafka                                          
✖ Linting
...
  55 errors

awesome-doge commented:
no license document

![截圖 2023-05-31 上午2 12 34](https://github.com/sindresorhus/awesome/assets/54278075/f1c94dc9-a2bc-47d6-b150-331b9d7c75f4)

vordimous commented:
@sindresorhus @asapelkin @awesome-doge I have sent in a PR to the maintainer addressing the issues https://github.com/semantalytics/awesome-kafka/pull/22

awesome-doge commented:
> @sindresorhus @asapelkin @awesome-doge I have sent in a PR to the maintainer addressing the issues [semantalytics/awesome-kafka#22](https://github.com/semantalytics/awesome-kafka/pull/22)

@vordimous 

Hello, there are still the following problems in your PR.

```
 ~/github/awesome-kafka │ on master ⇡8  npx awesome-lint                                                                                                                                                                                 ✔ │ at 02:38:00 PM
✖ Linting

  README.md:1:1
  ✖    1:1   The repository should have a description                      remark-lint:awesome-github
  ✖    1:1   License was not detected by GitHub                            remark-lint:awesome-github
  ✖    1:1   Missing or invalid Table of Contents                          remark-lint:awesome-toc
  ✖   12:3   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   13:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   14:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   15:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   16:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   17:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   18:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   19:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   20:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   21:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   22:3   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   23:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   24:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   25:5   Invalid list item link URL                                    remark-lint:awesome-list-item
  ✖   56:50  List item description must end with proper punctuation        remark-lint:awesome-list-item
  ✖  110:89  List item description must start with valid casing            remark-lint:awesome-list-item
  ✖  117:73  List item description must end with proper punctuation        remark-lint:awesome-list-item
  ✖  148:46  List item link and description must be separated with a dash  remark-lint:awesome-list-item
  ✖  150:55  List item link and description must be separated with a dash  remark-lint:awesome-list-item
  ✖  162:43  List item link and description must be separated with a dash  remark-lint:awesome-list-item

  23 errors
```

vordimous commented:
@awesome-doge All lint rules are fixed.

awesome-doge commented:
@vordimous 

It looks like you still have two issues that need to be fixed.

```
 ~/github/awesome-kafka │ on master ⇡9  npx awesome-lint                                                                        ✔ │ at 10:13:52 PM
✖ Linting

  README.md:1:1
  ✖  1:1  The repository should have a description  remark-lint:awesome-github
  ✖  1:1  License was not detected by GitHub        remark-lint:awesome-github

  2 errors
```

vordimous commented:
@sindresorhus  @awesome-doge All of the linter rules are passing. How can I trigger the workflow again?

```
% npx awesome-lint https://github.com/semantalytics/awesome-kafka
✔ Linting
```

vordimous commented:
@sindresorhus  @awesome-doge All of the linter rules are passing and the checks are passing.

vordimous commented:
@zacharywhitley @sindresorhus @awesome-doge bump

sindresorhus commented:
> You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. Just commenting “looks good” or simply marking the pull request as approved does not count! You have to actually point out mistakes or improvement suggestions.

sindresorhus commented:
> Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
